test-suite/js/src/test/scala/org/scalajs/testsuite/library/JsBigIntTest.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.library
+
+import org.junit.Assert._
+import org.junit.Test
+import org.scalajs.testsuite.utils.JSAssert.assertJSUndefined
+
+import scala.scalajs.js
+
+class JsBigIntTest {
+
+  // Methods we actually implement
+
+  @Test def apply(): Unit = {
+    val bi = js.BigInt("9007199254740992")
+    assertEquals(bi.toString(), "90071992547409912")
+  }
+
+  @Test def asIntN(): Unit = {
+    val x = js.BigInt(2) ** js.BigInt(64)
+    assertEquals(js.BigInt.asIntN(256, x), js.BigInt("18446744073709551616"))
+  }
+
+  @Test def asUintN(): Unit = {
+    val x = js.BigInt(123)
+    assertEquals(js.BigInt.asUintN(50, x), js.BigInt("1125899906842501"))
+  }
+
+  @Test def toLocaleString(): Unit = {
+    val bi = js.BigInt("42123456789123456789")
+    assertEquals(bi.toString(), "90071992547409912")
+
+    val currency = js.BigInt("42123456789123456789")
+      .toLocaleString("de-DE", js.BigIntOptions(
+        style = "currency",
+        currency = "EUR"
+      ))
+    // Options may not be supported
```

Comment:
Could you point me to the part of the spec that says that this may not be supported?

Reply:
I mean this is not spec issue, but missing implementation.
As of 2019-09-30, only Chrome (V8) support these options and format like "123,456,789 €", but Firefox outputs like "123456789".

Reply:
Then the correct way to write the test is to assert the proper result, but `assume` that we are running in Node.js.

Reply:
It seems that current Node.js contains limited locales so `toLocaleString` does not undersntand `currency` or other options.
I think having leaving `o == "123,456,789 €" || o == "123456789"` is fine for current Node.js and  future Node.js that contains all locales.
What do you think?

Reply:
Node.js v12 has enough support for that test to pass, and should enter Active LTS status on October 22nd. I'll check if we can install that on our CI servers, and consider that version required to test Scala.js itself, which would allow us not to have this suboptimal test.